v0/destinations/salesforce/transform.js

```diff
@@ -2,171 +2,169 @@ const get = require("get-value");
-  if (destination.Config.customMappings) {
-    destination.Config.customMappings.forEach(mapping => {
-      obj[mapping.from] = mapping.to;
-    });
-  }
-  const keys = Object.keys(obj);
-  keys.forEach(key => {
-    params[obj[key]] = get(message.properties, key);
-  });
-  return params;
+  const authUrl = `${SF_TOKEN_REQUEST_URL}?username=${
+    destination.Config.userName
+  }&password=${encodeURIComponent(
+    destination.Config.password
+  )}${encodeURIComponent(destination.Config.initialAccessToken)}&client_id=${
+    destination.Config.consumerKey
+  }&client_secret=${destination.Config.consumerSecret}&grant_type=password`;
+  const response = await axios.post(authUrl, {});
+
+  return {
+    token: `Bearer ${response.data.access_token}`,
+    instanceUrl: response.data.instance_url
+  };
 }
 
 // Basic response builder
 // We pass the parameterMap with any processing-specific key-value prepopulated
 // We also pass the incoming payload, the hit type to be generated and
 // the field mapping and credentials JSONs
-async function responseBuilderSimple(
-  message,
-  mappingJson,
-  ignoreMapJson,
-  destination,
-  targetEndpoint,
-  authorizationData
-) {
+function responseBuilderSimple(message, targetEndpoint, authorizationData) {
   // First name and last name need to be extracted from the name field
-  // and inserted into the message
-  let firstName = "";
-  let lastName = "";
-  const traits = getFieldValueFromMessage(message, "traits");
-  if (traits.name) {
-    // Split by space and then take first and last elements
-    const nameComponents = traits.name.split(" ");
-    firstName = nameComponents[0]; // first element
-    lastName = nameComponents[nameComponents.length - 1]; // last element
-    // Insert first and last names separately into message
-    traits.firstName = firstName;
-    traits.lastName = lastName;
-  }
-
-  const rawPayload = constructPayload(message, mappingJson);
-
-  if (!rawPayload.LastName || rawPayload.LastName.trim() === "") {
-    rawPayload.LastName = "n/a";
-  }
-
-  if (!rawPayload.Company || rawPayload.Company.trim() === "") {
-    rawPayload.Company = "n/a";
-  }
-
-  // Remove keys with undefined values
-  const payload = removeUndefinedValues(rawPayload);
-
-  // Get custom params from destination config
-  let customParams = getParamsFromConfig(message, destination);
-  customParams = removeUndefinedValues(customParams);
-
-  const customKeys = Object.keys(traits);
-  customKeys.forEach(key => {
-    if (
-      !defaultTraits.some(function(k) {
-        return ~k.indexOf(key);
-      }) &&
-      !ignoreMapJson.includes(key)
-    ) {
-      const val = traits[key];
-      if (val) {
-        payload[`${key}__c`] = val;
-      }
+  // get traits from the message
+  let traits = getFieldValueFromMessage(message, "traits");
+  // adjust for firstName and lastName
+  traits = { ...traits, ...getFirstAndLastName(traits, "n/a") };
```

Comment:
check if traits is defined and object before deconstructing it

Reply:
Done